app/src/main/java/br/com/dionataferraz/vendas/transaction/remote/TransactionApiDataSource.kt

```diff
@@ -0,0 +1,53 @@
+package br.com.dionataferraz.vendas.transaction.remote
+
+import br.com.dionataferraz.vendas.login.data.remote.ErrorModel
+import br.com.dionataferraz.vendas.login.data.remote.Result
+import br.com.dionataferraz.vendas.login.data.remote.RetrofitNetworkClient
+import br.com.dionataferraz.vendas.login.data.request.NewUserRequest
+import br.com.dionataferraz.vendas.login.data.response.UserResponse
+import br.com.dionataferraz.vendas.transaction.data.TransactionRequest
+import br.com.dionataferraz.vendas.transaction.data.TransactionResponse
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+class TransactionApiDataSource {
+
+    private val service = RetrofitNetworkClient
+        .createNetworkClient()
+        .create(TransactionApi::class.java)
+
+    suspend fun findAllByIdUser(idUser: Int): Result<List<TransactionResponse>, ErrorModel> {
+        return withContext(Dispatchers.IO) {
+            try {
+                val user = service.findAll(idUser.toString())
+                Result.Success(user)
+            } catch (exception:Exception){
+                println(exception)
+                println(exception.message)
+                println(exception.cause)
+                Result.Error(ErrorModel)
+            }
+        }
+    }
+
+    suspend fun register(idUser: Int, transactionRequest: TransactionRequest) {
+        return withContext(Dispatchers.IO) {
+            try {
+                service.register(idUser.toString(), transactionRequest)
```

Comment:
Se eu não me engano nós conseguimos utilizar o `@Path` como um `Int`
```suggestion
                service.register(idUser, transactionRequest)
```